openrndr-core/src/main/kotlin/org/openrndr/shape/ShapeContour.kt

```diff
@@ -592,51 +665,57 @@ data class ShapeContour(val segments: List<Segment>, val closed: Boolean, val po
                 val left = segments.last()
                 val right = segments.first()
                 fixedSegments.add(
-                    Segment(
-                        left.start,
-                        left.control,
-                        right.start
-                    )
+                        Segment(
+                                left.start,
+                                left.control,
+                                right.start
+                        )
                 )
             } else {
                 fixedSegments.add(segments.last())
             }
         }
         return ShapeContour(
-            if (segments.size > 1) fixedSegments else segments,
-            closed,
-            polarity
+                if (segments.size > 1) fixedSegments else segments,
+                closed,
+                polarity
         )
     }
 
+    /** Applies a boolean union operation between the [ShapeContour] and a [Shape]. */
     @Suppress("unused")
     fun union(other: Shape): Shape = union(this.shape, other)
 
+    /** Applies a boolean difference operation between the [ShapeContour] and another [Shape]. */
     @Suppress("unused")
     fun difference(other: Shape): Shape = difference(this, other)
 
+    /** Applies a boolean intersection operation between the [ShapeContour] and a [Shape]. */
     @Suppress("unused")
     fun intersection(other: Shape): Shape = intersection(this, other)
 
+    /** Calculates a [List] of all points of where paths intersect between the [ShapeContour] and a [Segment]. */
```

Comment:
Returns a [List] with all intersection points between ... ? 

Same in the next ones?